pandas/core/series.py

```diff
@@ -3075,6 +3076,15 @@ def _try_cast(arr, take_fast_path):
 
         subarr = maybe_cast_to_datetime(subarr, dtype)
 
+    elif isinstance(data, range):
+        # GH 16804
+        start, stop, step = _get_range_parameters(data)
+        if is_extension_type(dtype):
```

Comment:
What case does this handle?  Doesn't seem to be a matching test

Reply:
This is for dtypes that are specific to pandas that numpy doesn't understand, e.g. `pd.Series(range(5), dtype='category')`.  Will add a test for this when I fix the failing tests.